### PR TITLE
[Backport stable/8.5] feat: allow overriding default maxInboundMetadataSize through config and change default to 16KB

### PR DESCRIPTION
**Pull request by mustafadagher**
# Description
Backport of #20002 to `stable/8.5`.

relates to #19829 #11284 #19591

### CODE REVIEW

**Comment by mustafadagher** [comment]
I have no clue what happened with this one https://github.com/camunda/camunda/pull/20044

**Review by korthout — DISMISSED** [review]
❌ The tests for `spring-boot-starter-camunda-sdk` are missing. As the code is so different, we need to test this in order to release it.

💭 That will only be needed for 8.5, as the spring-sdk only exists since 8.5

**Review by remcowesterhoud — APPROVED** [review]

**Comment by remcowesterhoud** [comment]
I cannot merge this yet as it requires approval from @korthout

**Pull request merged**
